_posts/multirepo_comments.adoc

```diff
@@ -0,0 +1,96 @@
+
+## The Story
+
+I've seen this happening multiple times: you and a few teammates are starting a new project. The project is some complicated system that comprises several processes. You `git init` a new repo for each process without thinking about it too much, it feels natural. 
+
+As the system evolves, the processes start talking to each other over RPC, and since the processes use different programming languages, you use something like `protobuf` or `thrift`. Since the `.proto/.thrift` files are used by multiple components, they don't belong in any of the repos you created, so you create a new git repo and stick the shared API stuff in it.
+
+Since you're a great software engineer you always write unit tests for your code, and that fits snugly in your component's repo. A few weeks later, the system is a bit more mature and you want to start adding integration testing between components and system level testing to test the entire thing. But that doesn't really fit in any component's repo, and it's certainly not a shared-api thingy, so you create another repo for system testing.
+
+Later on you realize that two of the components can share some piece of infrastructure code, so you create a library out of it and put that in its own repo.
+
+Then you want some "magic script" to build your entire system from scratch and produce a shipable product, so you write it up. But where do you put it? Doesn't seem to fit in any of the repos you created... So you create a new repo for build stuff.
+
+Without noticing, you've just created ~10 git repositories for a single product. Welcome to MultiRepo Hell.
+
+While the above example sound extreme, I've actually seen this happen more than once. We'll now examine why this is a terrible way to go.
+
+## MultiRepo Hell
+
+There are a lot of consequences to going MultiRepo, so the list below is not complete.
+
+.Versioning
+Unless your APIs are simple, stable and well defined, you're going to need a way to specify which version process A knows how to talk to which version of process B. This means that *every git repo must be versioned in a sensible way*. And while "sensible" sounds simple enough, it is not (and that might deserve its own blog post). In a nutshell, by sensible I mean that you should be able to specify a dependency on some feature, some bugfix, or some API version. This quickly becomes a headache as developers must keep updating their component's version. When they forget to do so, the entire system might be in an inconsistent state, where components do not agree on which version of some other component they want to use.
+
+Multirepo creates a ton of interdependencies that simply do not exist in Monorepo, since in monorepo there is only one state of the code - the current commit - and not some combinatorial monster that depends on the exact state of each repo.
```

Comment:
I will try to trim it down. I think that a general comment for myself is "stop feeding the reader with a spoon and assume she can understand the idea from shorter descriptions of the problem". Would you agree with this statement? Or is it better to be explicit?